ci/travis/ci.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+{ SHELLOPTS_STACK="${SHELLOPTS_STACK-}|$(set +o); set -$-"; } 2> /dev/null  # Push caller's shell options (quietly)
+
+unset -f cd  # Travis defines this on Mac for RVM, but it floods the trace log and isn't relevant for us
+
+set -eo pipefail && if [ -n "${OSTYPE##darwin*}" ]; then set -ux; fi  # some options interfere with Travis's RVM on Mac
+
+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)"
+WORKSPACE_DIR="${ROOT_DIR}/../.."
+
+should_run_job() {
```

Comment:
top level comment on what this function do?
let's do this https://google.github.io/styleguide/shellguide.html#function-comments